woocommerce/woocommerce-ios
# [Mobile Payments] Specific alert provider for built-in reader connections

Part of: #8082 


## Description


In sTAP Away, we're adding support for Apple's built in card readers to take in-person payments. We continue to use Stripe's SDK to handle payments.

Stripe's connection flow is technically very similar to the bluetooth reader flow, but conceptually, there are differences and it's important that we make it clear to users that they are using the built in reader, and don't give the wrong impression that they are connecting to a bluetooth reader, to avoid confusion.

This PR adds an alert provider specialised for connection to a built-in reader, along with new alert view models and temporary text and images for the scanning and connecting stages of the connecting flow.

It does not handle error states, or "software updates" – these will be done in a future PR.

There are also new protocols, splitting up the alerts so that when we refactor the two connection controllers to share duplicated code, they can work with the alert providers as opaque types.

## Testing instructions


Using an iPhone (XS or above on iOS 15.4 or newer, with a passcode set and signed in to iCloud)

1. Navigate to `Menu > Settings > Experimental features`
2. Turn on `Tap to Pay on iPhone`
3. Navigate to `Menu > Payments > Collect payment`
4. Go through the payment flow, and select `Card` on the payment method screen
5. When asked for a reader type, tap `Tap to Pay on iPhone` and go through the Terms of Service Apple ID linking (if you've not done so before)
6. Observe that specific messages for the built in reader are shown during the connecting process (note that images and text are all placeholders!)

## Screenshots


https://user-images.githubusercontent.com/2472348/205371624-1190c6fe-13f6-47c1-b020-62915ab2e9ad.mp4



---
